### PR TITLE
Example for Koa 2 and option to override content-type

### PR DESCRIPTION
**Pull request by niieani**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -98,7 +99,7 @@ function send(ctx, path, opts) {
     ctx.set('Last-Modified', stats.mtime.toUTCString());
     ctx.set('Content-Length', stats.size);
     ctx.set('Cache-Control', 'max-age=' + (maxage / 1000 | 0));
-    ctx.type = type(path);
+    ctx.type = typeOverride ? typeOverride : type(path);
     ctx.body = fs.createReadStream(path);
```
Comment (coderhaoxin) [inline index.js]: Overriding the `content type` for the `whole` served directory?
If you want to set `contnt-type` for special file, you can

```js
yield send(this, this.path);
if (...) {
  this.type = 'some'
}
```

File: Readme.md [hunk 2]
```diff
@@ -48,6 +49,25 @@ app.use(function *(){
 })
 ```
 
+## Example (Koa@^2.0) with await/async
+
+```js
+var send = require('koa-send');
+var koa = require('koa');
+var app = koa();
+
```
Comment (coderhaoxin) [inline Readme.md]: For koa2, it should be.

```js
var Koa = require('koa');
var app = new Koa();
```

**Comment by niieani** [comment]
Overriding the content type for the whole served directory does make sense in certain use cases. Mine was to support [LetsEncrypt](https://github.com/letsencrypt/letsencrypt)'s verification handshake, which requires all the files in the verification directory to be served with "text/plain" content type.
I've corrected the Readme regarding Koa@2. I've originally missed correcting the `require`' area, since I'm using the new `import` syntax locally.

**Comment by coderhaoxin** [comment]
> Overriding the content type for the whole served directory does make sense in certain use cases.

For your use cases, just need
```js
yield send(...);
this.type = 'some';
```

**Comment by coderhaoxin** [comment]
Anyway, `Example for Koa 2` and `option to override content-type` should be two different PRs. >_<

**Comment by coderhaoxin** [comment]
@niieani `Example for Koa 2` landed 35cced5e1183f76dc6367b6944089db5385eddc7

File: index.js (hunk 1 above)
Comment (niieani) [inline index.js]: How? Can you give a proper example? Because as far as I see, `ctx.type` is always overwritten by koa-send.

**Pull request closed without merging**